### PR TITLE
chore(doc-gen, docs-app): generate version for stable snapshot

### PR DESCRIPTION
**Pull request by Narretz**
The "stable snapshot" is based on the latest commit on the current stable
branch, i.e. a preview of the next patch version.

Next thing:
- update the Firebase deployment so that on the branch with distTag = latest gets deployed to docs.angularjs.org and as to gcs as snapshot-stable

### CODE REVIEW

File: docs/app/src/versions.js [hunk 1]
```diff
@@ -12,10 +12,16 @@ angular.module('versions', ['currentVersionData', 'allVersionsData'])
             /** @this VersionPickerController */
             function VersionPickerController($location, $window, CURRENT_NG_VERSION, ALL_NG_VERSIONS) {
 
-      var versionStr = CURRENT_NG_VERSION.isSnapshot ? 'snapshot' : CURRENT_NG_VERSION.version;
+      var versionStr = CURRENT_NG_VERSION.version;
+
+      if (CURRENT_NG_VERSION.isSnapshot) {
+        versionStr = CURRENT_NG_VERSION.distTag === 'latest' ? 'snapshot-stable' : 'snapshot';
```
Comment (petebacondarwin) [inline docs/app/src/versions.js]: perhaps we could rename `snapshot` and `snapshot-stable` to `next` and `stable` respectively?

File: lib/versions/version-info.js [hunk 2]
```diff
@@ -231,3 +232,4 @@ if (NO_REMOTE_REQUESTS) {
 
 console.log('CDN version (' + versionSource + '):', cdnVersion ? cdnVersion.raw : 'No version found.');
 console.log('Current version (' + versionSource + '):', exports.currentVersion.raw);
+console.log('Latest Stable', exports.latestStable);
```
Comment (petebacondarwin) [inline lib/versions/version-info.js]: `exports.latestStable` is never assigned in this file....

File: docs/app/src/versions.js (hunk 1 above)
Comment (Narretz) [reply docs/app/src/versions.js]: I don't know. I think it's important that these are snapshots of a branch. "Stable" says "release" to me, not snapshot. "Next" is better, but it also doesn't say it's a snapshot. What about "snapshot-next"?

Comment (Narretz) [reply docs/app/src/versions.js]: btw, these are not the labels that are shown - for "snapshot" we show "master", and for "snapshot-stable" we show "v1.x.y+1-snapshot"

Comment (petebacondarwin) [reply docs/app/src/versions.js]: I see. Cool to leave them as-is then.

Comment (petebacondarwin) [reply docs/app/src/versions.js]: or `master-snapshot` and `stable-snapshot`

File: docs/config/processors/versions-data.js [hunk 3]
```diff
@@ -112,6 +129,20 @@ module.exports = function generateVersionDocProcessor(gitData) {
...
+        var semverObj = semver.parse(version);
+        return semverObj === null ? false : !semverObj.prerelease.length;
+      }
+
+      function createSnapshotStableLabel(version) {
+        var increment = Number(version.slice(-1)) + 1;
+
+        var label = 'v' + (version.slice(0, -1) + increment) + '-snapshot';
+
+        return label;
```
Comment (petebacondarwin) [inline docs/config/processors/versions-data.js]: would it not be safer to use semver to do this massaging of the version?

File: docs/app/src/versions.js (hunk 1 above)
Comment (Narretz) [reply docs/app/src/versions.js]: The labels in the version picker are now "master-snapshot" and e.g. "v1.6.7-snapshot"

**Review by petebacondarwin — APPROVED** [review]
Great!
It is a shame that we have the same logic to compute the `snapshot` and `snapshot-stable` version strings repeated in both the dgeni processor and the application code. But I think we can live with that.

File: docs/config/processors/versions-data.js [hunk 4]
```diff
@@ -112,6 +129,20 @@ module.exports = function generateVersionDocProcessor(gitData) {
...
+      // https://github.com/kaelzhang/node-semver-stable/blob/34dd29842409295d49889d45871bec55a992b7f6/index.js#L25
+      function semverIsStable(version) {
+        var semverObj = semver.parse(version);
+        return semverObj === null ? false : !semverObj.prerelease.length;
+      }
+
+      function createSnapshotStableLabel(version) {
+        var semverObj = semver.parse(version);
+
+        var label = 'v' + semverObj.inc('patch').version + '-snapshot';
```
Comment (gkalpak) [inline docs/config/processors/versions-data.js]: It would be nice if we showed `v1.6.x-snapshot` (instead of `v1.6.7-snapshot`, which is kind of unclear).

Comment (Narretz) [reply docs/config/processors/versions-data.js]: Yeah, I guess that's clearer

**Pull request merged**